combat-trainer.lic

```diff
@@ -2713,12 +2719,16 @@ class AttackProcess
   end
 
   def shoot_aimed(command, game_state)
+    echo("Time now: #{Time.now.to_i}") if $debug_mode_ct
+    echo("@firing_timer: #{@firing_timer}") if $debug_mode_ct
+    echo("@firing_delay: #{@firing_delay}") if $debug_mode_ct
     case bput(command, 'isn\'t loaded', 'There is nothing', 'But your', /you (fire|poach|snipe) an? (?:.*\s)?(arrows?|bolt|shard|rock|sphere|clump|coral|fist|holder|lump|patella|pellet|pulzone|quadrello|quill|stone|stopper|verretto|crumb|spine|mantrap spike|dragon|icicle|fang|scale|grey-black spike|bacon strip|page|naga|thorn).* at/i, 'I could not find', 'with no effect and falls to the ground', 'Face what', 'How can you (poach|snipe)', 'you don\'t feel like fighting right now', 'That weapon must be in your right hand to fire')
     when /How can you (poach|snipe)/
       shoot_aimed('shoot', game_state)
     when /you (fire|poach|snipe) an? (?:.*\s)?(arrows?|bolt|shard|rock|sphere|clump|coral|fist|holder|lump|patella|pellet|pulzone|quadrello|quill|stone|stopper|verretto|blowgun dart|crumb|spine|mantrap spike|dragon|icicle|fang|scale|grey-black spike|bacon strip|page|naga|thorn).* at/i
       game_state.action_taken
       ammo = Regexp.last_match(2)
+      @firing_check = 0
```

Comment:
❓ I think you can get rid of `@firing_check` if you set `@firing_timer = Time.now` here (and get rid of line 2848).

What do you think?

Reply:
The `@firing_check` catches the loop because if it is the second time in we do not want `@firing_timer` to change.

Reply:
How much time passes between this line and line 2849?